### PR TITLE
Testsuite: pass -i argument to runghc invocations

### PR DESCRIPTION
**Pull request by sheaf**
This commit modifies the generation of runghc commands to pass an additional `-i` argument to account for the change in working directory.

This ensures that, in the testsuite, `runghc` is able to see other modules. For example, an invocation of `runghc` to run a Custom Setup script will now properly see the modules imported, e.g. if one has a directory structure like

```
  test.cabal
  Setup.hs
  SetupDep.hs
```

and `Setup.hs` imports `SetupDep.hs`.

### CODE REVIEW

File: cabal-testsuite/src/Test/Cabal/Script.hs [hunk 1]
```diff
@@ -77,14 +77,19 @@ runghc senv mb_cwd env_overrides script_path args = do
...
-runnerCommand senv _mb_cwd _env_overrides script_path args = do
+runnerCommand senv mb_cwd _env_overrides script_path args = do
     (prog, _) <- requireProgram verbosity runghcProgram (runnerProgramDb senv)
     return (programPath prog,
             runghc_args ++ ["--"] ++ map ("--ghc-arg="++) ghc_args ++ [script_path] ++ args)
   where
     verbosity = runnerVerbosity senv
     runghc_args = []
-    ghc_args = runnerGhcArgs senv
+    ghc_args = runnerGhcArgs senv ++ srcArg
```
Comment (mpickering) [inline cabal-testsuite/src/Test/Cabal/Script.hs:87]: I think the more precise way to achieve this is to pass cwd to `runnerGhcArgs` and modify `ghcOptSourcePath`.

I do note though that there `ghcOptSourcePathClear` is set, which seems at odds with this change.. what is `mb_cwd` typically set to?

Comment (sheaf) [reply cabal-testsuite/src/Test/Cabal/Script.hs:87]: It's typically set to the path to the test in the test-suite, see e.g. this code in `Test.Cabal.Prelude.setup''`:

```haskell
  do r <- liftIO $ runghc (testScriptEnv env)
                          (Just (testCurrentDir env))
                          (testEnvironment env)
                          (testCurrentDir env </> prefix </> "Setup.hs")
                          (NE.toList full_args)
 ```

**Review by geekosaur — APPROVED** [review]

**Comment by Mikolaj** [comment]
How can we help move this PR along? IIRC, it's needed to compile/test https://github.com/haskell/cabal/pull/9551 right?

**Comment by Mikolaj** [comment]
@mpickering: do you think you could approve this PR?

**Comment by sheaf** [comment]
> @mpickering: do you think you could approve this PR?

I still need to address his comment. I will mark as Draft for now.

**Comment by sheaf** [comment]
> > @mpickering: do you think you could approve this PR?
> 
> I still need to address his comment. I will mark as Draft for now.

This PR is now good to go.

**Review by mpickering — APPROVED** [review]

**Comment by Kleidukos** [comment]
@mergifyIO backport 3.12

**Pull request merged**